### PR TITLE
feat(plugin): NoReplyPings

### PR DESCRIPTION
**Pull request by MrDiamondDog**
Doesn't mention you when someone who left mentions on replies to you

### CODE REVIEW

File: src/plugins/noReplyPings/index.ts [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+export default definePlugin({
+    name: "NoReplyPings",
+    description: "Disables getting pinged by replies, even if the sender didn't turn it off",
+    authors: [Devs.MrDiamond],
+    patches: [
+        {
+            find: "_channelMessages",
+            replacement: {
+                match: /receiveMessage\((\i)\)\{/,
+                replace: "$&if($self.isReplyMention($1))$1.mentions=[];"
```
Comment (Vendicated) [inline src/plugins/noReplyPings/index.ts:30]: you should not remove all mentions. you should only remove your own mention

File: src/plugins/noReplyPings/index.ts [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+ * Copyright (c) 2023 Vendicated and contributors
+ * SPDX-License-Identifier: GPL-3.0-or-later
+ */
+
+import { Devs } from "@utils/constants";
+import definePlugin from "@utils/types";
+import { MessageStore, UserStore } from "@webpack/common";
+import { Message } from "discord-types/general";
+
+// for some reason discord has two different types for messages, isn't that just amazing design
```
Comment (Vendicated) [inline src/plugins/noReplyPings/index.ts:12]: - api message (raw json from gateway)
- local message (serialised & proper class)

Comment (MrDiamondDog) [reply src/plugins/noReplyPings/index.ts:12]: Changed to `MessageJSON` type which is the api message

**Comment by ant0n-0x0000** [comment]
This has been updated in PR #1975

**Pull request closed without merging**